subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -161,4 +171,54 @@ public boolean isSatisfiedBy(Task element) {
 
         // TODO - check should depend on test
     }
+
+    private void configureTestedComponent(final Project project) {
+        project.getPlugins().withType(SwiftExecutablePlugin.class, new Action<SwiftExecutablePlugin>() {
+            @Override
+            public void execute(SwiftExecutablePlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+
+        project.getPlugins().withType(SwiftLibraryPlugin.class, new Action<SwiftLibraryPlugin>() {
+            @Override
+            public void execute(SwiftLibraryPlugin plugin) {
+                configureTestedSwiftComponent(project);
+            }
+        });
+    }
+
+    private void configureTestedSwiftComponent(Project project) {
+        TaskContainer tasks = project.getTasks();
+        DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
+
+        SwiftCompile compile = tasks.create("compileMainSwiftWithTestingEnabled", SwiftCompile.class, configureCompileTestedComponentTask(tasks.withType(SwiftCompile.class).getByName("compileSwift")));
```

Comment:
I don't think we should add another compilation task. Instead, we should make the debug variant unit testable by default, by adding the relevant arguments to the existing `compileSwift` task (which represents the debug variant).

When we add the release variant (soon), it would not be built with this flag, meaning you'll be able to run functional tests but not unit tests against the release variant.